package_bag/tests.py

```diff
@@ -125,36 +119,94 @@ def test_discover_bags(self):
         self.assertIn("Error processing discovered bag", str(exc.exception))
         self.assertEqual(len(listdir(settings.TMP_DIR)), valid_bags)
 
+    def tearDown(self):
+        for d in [settings.TMP_DIR, settings.SRC_DIR, settings.DEST_DIR]:
+            if isdir(d):
+                shutil.rmtree(d)
+
+
+class TestRightsAssigner(TestCase):
+    fixtures = ["get_rights.json"]
+
+    def setUp(self):
+        with open(join(RIGHTS_FIXTURE_DIR, 'rights_service_response.json')) as json_file:
+            self.rights_service_response = json.load(json_file)
+        set_up_directories([settings.TMP_DIR, settings.SRC_DIR, settings.DEST_DIR])
+
     @patch('package_bag.routines.post')
-    def test_get_rights(self, mock_rights):
+    def test_run(self, mock_rights):
         """Ensures that rights are correctly retrieved and assigned."""
-        add_bags_to_db(settings.TMP_DIR, self.expected_count)
         mock_rights.return_value.status_code = 200
         mock_rights.return_value.json.return_value = self.rights_service_response
-        assign_rights = RightsAssigner().run()
-        mock_rights.assert_called_with(
-            'http://aquila-web:8000/rights',
-            json={'identifiers': RIGHTS_ID, 'start_date': None, 'end_date': END_DATE})
-        self.assertIsNot(False, assign_rights)
+        for bag in Bag.objects.filter(process_status=Bag.DISCOVERED):
+            assign_rights = RightsAssigner().run()
+            mock_rights.assert_called_with(
+                'http://aquila-web:8000/rights',
```

Comment:
I think it might make sense to use a config value here (since that's where this should be coming from) rather than hardcoding a value?